cpu/efm32/periph/timer.c

```diff
@@ -101,68 +138,176 @@ int timer_init(tim_t dev, unsigned long freq, timer_cb_t callback, void *arg)
+        }
+    }
 
     return 0;
 }
 
 int timer_clear(tim_t dev, int channel)
 {
-    timer_config[dev].timer.dev->CC[channel].CTRL = _TIMER_CC_CTRL_MODE_OFF;
+    if (!_is_letimer(dev)) {
+        TIMER_TypeDef *tim = timer_config[dev].timer.dev;
+        tim->CC[channel].CTRL = _TIMER_CC_CTRL_MODE_OFF;
+    }
+    else {
+        LETIMER_TypeDef *tim = timer_config[dev].timer.dev;
+        switch (channel)
+        {
+            case 0:
+                LETIMER_IntDisable(tim, LETIMER_IEN_COMP0);
+                LETIMER_IntClear(tim, LETIMER_IFC_COMP0);
+                break;
+
+            case 1:
+                LETIMER_IntDisable(tim, LETIMER_IEN_COMP1);
+                LETIMER_IntClear(tim, LETIMER_IFC_COMP1);
+                break;
+            default:
+                return -1;
+        }
+    }
     return 0;
 }
 
 unsigned int timer_read(tim_t dev)
 {
-    return (unsigned int) TIMER_CounterGet(timer_config[dev].timer.dev);
+    if (_is_letimer(dev)) {
+        /* LETIMER is countdown only, so we invert the value */
+        return (unsigned int) 0xffff
+                    - LETIMER_CounterGet(timer_config[dev].timer.dev);
+    }
+    else {
+        return (unsigned int) TIMER_CounterGet(timer_config[dev].timer.dev);
+    }
 }
 
 void timer_stop(tim_t dev)
 {
-    TIMER_Enable(timer_config[dev].timer.dev, false);
+    if (_is_letimer(dev)) {
+        LETIMER_TypeDef *tim = timer_config[dev].timer.dev;
+        if (tim->STATUS & LETIMER_STATUS_RUNNING) {
+            pm_unblock(EFM32_LETIMER_PM_BLOCKER);
+        }
+        LETIMER_Enable(timer_config[dev].timer.dev, false);
+    }
+    else {
+        TIMER_TypeDef *tim = timer_config[dev].timer.dev;
+        if (tim->STATUS & TIMER_STATUS_RUNNING) {
+            pm_unblock(EFM32_TIMER_PM_BLOCKER);
+        }
+        TIMER_Enable(timer_config[dev].timer.dev, false);
+    }
 }
 
 void timer_start(tim_t dev)
 {
-    TIMER_Enable(timer_config[dev].timer.dev, true);
+    if (_is_letimer(dev)) {
+        LETIMER_TypeDef *tim = timer_config[dev].timer.dev;
+        if (!(tim->STATUS & LETIMER_STATUS_RUNNING)) {
+            pm_block(EFM32_LETIMER_PM_BLOCKER);
+        }
+        LETIMER_Enable(timer_config[dev].timer.dev, true);
+    }
+    else {
+        TIMER_TypeDef *tim = timer_config[dev].timer.dev;
+        if (!(tim->STATUS & TIMER_STATUS_RUNNING)) {
+            pm_block(EFM32_TIMER_PM_BLOCKER);
+        }
+        TIMER_Enable(timer_config[dev].timer.dev, true);
+    }
 }
 
 #ifdef TIMER_0_ISR
 void TIMER_0_ISR(void)
 {
-    TIMER_TypeDef *tim = timer_config[0].timer.dev;
-
-    for (int i = 0; i < (int) CC_CHANNELS; i++) {
-        if (tim->IF & (TIMER_IF_CC0 << i)) {
-            tim->CC[i].CTRL = _TIMER_CC_CTRL_MODE_OFF;
-            tim->IFC = (TIMER_IFC_CC0 << i);
-            isr_ctx[0].cb(isr_ctx[0].arg, i);
+    tim_t dev = 0;
+
+    if (_is_letimer(dev)) {
+        LETIMER_TypeDef *tim = timer_config[dev].timer.dev;
+
+        for (int i = 0; i < 2; i++) {
```

Comment:
Can this magic number be defined as well? This probably relates to #13900 as well.